### ISSUE TITLE
[Enhancement]: resource: aws_lb_target_group - Automatic Target Weights (ATW)

### ISSUE THREAD

**Issue by cwilliams-intelerad** [hashicorp/terraform-provider-aws#34644]
### Description

AWS load balancer target groups now have an automatic target weights (AWT) feature that added an allow a load balance type algorithm of "weighed random" with anomaly mitigation. I would like the feature to be supported in the aws_lb_target_group resource.

### Affected Resource(s) and/or Data Source(s)

aws_lb_target_group

### Potential Terraform Configuration

```terraform
resource "aws_lb_target_group" "alb-example" {
  name        = "tf-example-lb-alb-tg"
  port                          = 80
  protocol                      = "TCP"
  vpc_id                        = aws_vpc.main.id
  load_balancing_algorithm_type = "weighted_random"
  anomaly_mitigation            = true
}
```

### Would you like to implement a fix?

None

**Comment by bclodius** [comment]
@ewbankkit is Hashicorp open to contributions for this?

**Comment by ewbankkit** [comment]
@bclodius All contributions always welcome 🎅.
This issue has not be claimed.

**Comment by bclodius** [comment]
@ewbankkit I'd like to claim this!  Will take it up next week 🙏 .

**Comment by bclodius** [comment]
I'm making a good progress on a PR.  Should have something up Thursday December 28th.

**Comment by bclodius** [comment]
@cwilliams-intelerad @nitrocode FYI I've raised a PR here https://github.com/hashicorp/terraform-provider-aws/pull/35083

I opted to implement `load_balancing_anomaly_mitigation` with values `on` and `off` since that's what the AWS API supports.  I don't know why AWS didn't choose to just support `true` and `false`.  My guess is they want to future proof to support future values.  I think they realized this mistake when they made `load_balancing.cross_zone.enabled` with values `true`, `false`, and `use_load_balancer_configuration` 😄 .

From: https://docs.aws.amazon.com/cli/latest/reference/elbv2/modify-target-group-attributes.html

> load_balancing.algorithm.anomaly_mitigation - Only available when load_balancing.algorithm.type is weighted_random . Indicates whether anomaly mitigation is enabled. The value is on or off . The default is off .
